components/outsideAlerter.js

```diff
@@ -0,0 +1,34 @@
+import React, { useRef, useEffect } from "react";
```

Comment:
also this is pretty tight. found it on stack overflow which is a component that captures clidks OUTSIDE the component